ibis-server/app/model/metadata/trino.py

```diff
@@ -16,35 +13,45 @@
 
 
 class TrinoMetadata(Metadata):
-    def __init__(self, connection_info: TrinoConnectionInfo | ConnectionUrl):
+    def __init__(self, connection_info: TrinoConnectionInfo):
         super().__init__(connection_info)
+        self.connection = DataSource.postgres.get_connection(connection_info)
 
     def get_table_list(self) -> list[Table]:
         schema = self._get_schema_name()
-        sql = f"""SELECT
-                t.table_catalog,
-                t.table_schema,
-                t.table_name,
-                c.column_name,
-                c.data_type,
-                c.is_nullable
-              FROM
-                information_schema.tables t
-              JOIN
-                information_schema.columns c 
-              ON t.table_schema = c.table_schema 
-              AND t.table_name = c.table_name
-              WHERE
-                t.table_type IN ('BASE TABLE', 'VIEW')
-              AND t.table_schema NOT IN ('information_schema', 'pg_catalog')
-              AND t.table_schema = '{schema}'"""
+        sql = f"""
+                SELECT
+                    t.table_catalog,
+                    t.table_schema,
+                    t.table_name,
+                    c.column_name,
+                    c.data_type,
+                    c.is_nullable,
+                    c.column_comment
+                FROM
+                    information_schema.tables AS t
+                INNER JOIN
+                    information_schema.columns AS c
+                    ON t.table_catalog = c.table_catalog
+                    AND t.table_schema = c.table_schema
+                    AND t.table_name = c.table_name
+                WHERE t.table_schema = '{schema}'
```

Comment:
I didn't try it but maybe we can merge it with the table comments query? Something like
```sql
SELECT
    t.table_catalog,
    t.table_schema,
    t.table_name,
    c.column_name,
    c.data_type,
    c.is_nullable,
    c.column_comment,
    tc.comment
FROM
    information_schema.tables AS t
INNER JOIN
    information_schema.columns AS c
    ON t.table_catalog = c.table_catalog
    AND t.table_schema = c.table_schema
    AND t.table_name = c.table_name
INNER JOIN
    system.metadata.table_comments AS tc
    ON t.table_schema = tc.schema_name
    AND t.table_name = tc.table_name
WHERE t.table_schema = '{schema}'
```

Reply:
No. Trino does not implement `getTableHandle` for SystemMetadata. It will fail when executing.